frontend/src/components/custom-tools/prompt-card/PromptCardItems.jsx

```diff
@@ -179,101 +323,288 @@ function PromptCardItems({
+                                  {displayPromptResult(
+                                    result.find(
+                                      (r) => r?.profileManager === profileId
+                                    )?.output,
+                                    true
+                                  )}
+                                </div>
+                              </Typography.Paragraph>
+                            )}
+                            <div className="prompt-profile-run">
+                              <Tooltip title="Run">
+                                <Button
+                                  size="small"
+                                  type="text"
+                                  className="prompt-card-action-button"
+                                  onClick={() => handleRun(profileId, false)}
+                                  disabled={
+                                    isRunLoading[
+                                      `${selectedDoc?.document_id}_${profileId}`
+                                    ]
+                                  }
+                                >
+                                  <PlayCircleOutlined className="prompt-card-actions-head" />
+                                </Button>
+                              </Tooltip>
+                              <Tooltip title="Run All">
+                                <Button
+                                  size="small"
+                                  type="text"
+                                  className="prompt-card-action-button"
+                                  onClick={() => handleRun(profileId, true)}
+                                  disabled={
+                                    isRunLoading[
+                                      `${selectedDoc?.document_id}_${profileId}`
+                                    ]
+                                  }
+                                >
+                                  <PlayCircleFilled className="prompt-card-actions-head" />
+                                </Button>
+                              </Tooltip>
+                              <Tooltip title="Expand">
+                                <Button
+                                  size="small"
+                                  type="text"
+                                  className="prompt-card-action-button"
+                                  onClick={() => handleExpandClick(profile)}
+                                >
+                                  <ArrowsAltOutlined className="prompt-card-actions-head" />
+                                </Button>
+                              </Tooltip>
+                            </div>
+                          </div>
+                        </>
+                      </Col>
+                    </motion.div>
+                  );
+                })}
+            </AnimatePresence>
+            {singlePassExtractMode &&
+              (() => {
```

Comment:
@jagadeeswaran-zipstack Can this function be defined separately and then called from here? This will improve code readability.

Reply:
Sure